internal/epoch/epoch_manager.go

```diff
@@ -186,6 +186,10 @@ type Manager struct {
 	log      logging.Logger
 	timeFunc func() time.Time
 
+	// whether the underlying storage should be considered read only. Influences
+	// whether index compaction occurs automatically.
+	readOnly bool
```

Comment:
shouldn't this be exposed on the `blob.Storage` itself?

Reply:
Ignore the comment above, I had not seen #3230